QuizApp/QuizApp/UI/Screens/SolvingQuiz/ProgressView.swift

```diff
@@ -0,0 +1,69 @@
+import UIKit
+
+class ProgressView: UIView {
+
+    private var label: UILabel!
+    private var stackView: UIStackView!
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+
+        createViews()
+        styleViews()
+        defineLayoutForViews()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    func set(colors: [UIColor]) {
+        stackView.arrangedSubviews.forEach { $0.removeFromSuperview() }
+
+        for index in 0..<colors.count {
+            if colors[index] == UIColor.white {
+                label.text = "\(index + 1)/\(colors.count)"
```

Comment:
You are making a decision based on color. That is not correct. Instead, pass some variables to indicate what is the current questing index and what is the total number of questions.

Reply:
Solved on next branch